File: SantaFinder.Web/Areas/Auth/Models/ChangeProfile/ClientProfileChangeModel.cs
using SantaFinder.Entities;
using SantaFinder.Web.Models.Shared;

namespace SantaFinder.Web.Areas.Auth.Models.ChangeProfile
{
    public class ClientProfileChangeModel: IProfileChangeModel
    {
        public string Email { get; set; }
        public string Name { get; set; }
        public AddressWithLocation Address { get; set; }
        public string Password { get; set; }
        public NewPasswordModel NewPassword { get; set; }
    }
}

File: SantaFinder.Web/Client/src/app/data-services/view-models/change-profile/client-profile-change.model.ts
import { Location } from '../../../shared/models/location';
import { Address } from '../../../shared/models/address.model';

export interface ClientProfileChangeModel {
    email: string;
    name: string;
    address: {
        line: Address;
        location: Location
    };
    password: string;
    newPassword: {
        password: string;
        passwordConfirmation: string;
    };
}